Feat/2025 update

GiugAles opened the pull request:
Most controversial change is probably the introduction of the term "operation" in [line 189 of README.md](https://github.com/HSG-WAS-SS24/exercise-1/compare/feat/2025-update?expand=1#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R189). Please let me know if this makes sense or not. I changed it, because I was looking through the `.asl` file for the corresponding actions not realizing, that this comes from the environment. I used the term in the `README.md`, because I first saw it on @JeremyLemee poster regarding SEM and then realized that the implementation is also decorated with `@OPERATION`.

GiugAles committed:
docs(tutorial): reduce ambiguity

The term 'operation' is also used in the context of the Signifier
Exposure Mechanism to indicate interaction possibilities with
environment.

GiugAles committed:
docs(Troubleshooting): added fix to frozen console

GiugAles committed:
chore(gradlew): added +x on gradlew

When commiting on certain OS, file permissions are sometime lost.
However, gradlew needs executable permissions.

andreiciortea commented on README.md:
```diff
@@ -291,7 +291,7 @@ This will initiate the gold handling process and will eventually generate the tr
 The team of miners has elected a leader agent, which is responsible for keeping track of the depot coordinates and periodically moving the depot in the grid environment for security reasons. In the Mining World GUI, observe that the depot is moved to a new position every few seconds. You miner agent remains stuck carrying gold in the initial position of the depot — similar to the stuck Roomba you ...
 
-Thankfully, agents can communicate with one other to share knowledge about the environment. In AgentSpeack, plan bodies can contain _communication actions_, which allow agents to send messages. For example, the miner agent can perform the action `.send(tell, leader, sunny)` to tell the leader that it is `sunny`. The leader will then acquire a new `sunny` belief — and will also keep note that the  ...
+Thankfully, agents can communicate with one other to share knowledge about the environment. In AgentSpeack, plan bodies can contain _communication actions_, which allow agents to send messages. For example, the miner agent can perform the action `.send(leader, tell, sunny)` to tell the leader that it is `sunny`. The leader will then acquire a new `sunny` belief — and will also keep note that the  ...
```
Good catch!

andreiciortea commented on README.md:
```diff
@@ -216,7 +216,7 @@ The `@gold_perceived_plan` should only be applicable if the agent is ready to ex
 Currently, the miner agent can perceive gold nuggets but it cannot handle them.
 
-Your fourth task is to complete the `@handle_gold_plan` in [miner.asl](src/agt/miner.asl) to enable the agent to achieve its goal of collecting a perceived gold nugget and delivering it to the depot. The triggering event and the context of the plan are already provided, and there are two available actions that you can use in the plan body:
+Your fourth task is to complete the `@handle_gold_plan` in [miner.asl](src/agt/miner.asl) to enable the agent to achieve its goal of collecting a perceived gold nugget and delivering it to the depot. The triggering event and the context of the plan are already provided, and there are two available operations that you can use in the plan body. Operations are actions that are implemented by the env ...
```
One of the main objectives when writing this tutorial was to streamline the concepts and terminology as much as possible. Note that we don't talk about the environment as a first-class abstraction in this tutorial — and we don't mention artifacts either. Let's stick to actions. Suggestion: "are already provided, and the environment exposes two actions you can use in the plan body:". We can also mention when presenting the tutorial that these actions are implemented by the environment.

I will further detail the distinction between operations and actions on the PR comment thread.

andreiciortea replied:
@GiugAles: after reading the discussion on Slack with @JeremyLemee, I suggest leaving the original formulation about actions as it was and instead adding the following note after `pick` and `drop` are introduced:

> **Note (internal vs. external actions)**: The special actions we've seen so far (e.g., `.print`, `jia.random`) are called _internal actions_ and are typically used for programming convenience. In contrast, `pick` and `drop` are external actions: an action afforded by the environment and executed by an agent's effectors (or actuators). When an agent invokes an external action, it usually affects the environment—and other agents may also perceive it. An external action is what we usually think of as an agent action. External actions are not preceded by a prefix (e.g., `jia.` or `.` for the default prefix).

andreiciortea commented on README.md:
```diff
@@ -362,3 +362,17 @@ Now with your team of miners working at full force, you lay back, relax, and enj
 | :-- |
 
 
+## Troubleshooting
```
We need to scope and describe the issue more precisely. Suggestion:

> There is a known issue with this tutorial for MacOS users: in rare occurrences, when switching between the MAS console and the tutorial GUI, one of them might freeze. If you encounter this issue, we suggest disabling (...)

For consistency, make sure to add links to the relevant lines (just like we do in the rest of the README).

andreiciortea reviewed (changes requested):
Thanks @GiugAles! I left a few comments and suggestions. In addition:

- I've created a new organization for FS25; we should move this repo there (ideally with the new changes); I will send the link on Slack
- we need to udpate the contact person for this tutorial (currently, it is Danai)
- throughout the README, there are references to specific lines (e.g., in miner.asl), but the links point to the file; we should add links directly to the relevant lines
- one thing we did not get to do last year is to update the code to the (newly created) programming style guide for JaCaMo: programming style: https://jacamo-lang.github.io/documentation/programming-style/
- the build file includes `run` as a default gradle task, but this task does not exist; we should remove the default task — or add an existing one

GiugAles committed:
chore(miner, leader): apply JaCaMo programming style

Applied styleguide as described on
https://jacamo-lang.github.io/documentation/programming-style/. Chose
second variant for plans since it implies that actions will always
end with `;`.

GiugAles committed:
docs(README.md): updated mailto and file links

Note that markdown does not support line number links natively. This
depends on the renderer.

GiugAles committed:
docs(README.md): tweaking documentation

To not introduce the environment as first-class abstraction, refer to
operations as external actions, which is in line with
Multi-Agent-Oriented Programmin, chapter 4.2.

GiugAles committed:
build(build.gradle): setting task_1_4 as default

Task 'run' does not exist.
